src/library/scala/collection/mutable/RedBlackTree.scala

```diff
@@ -0,0 +1,502 @@
+package scala.collection.mutable
+
+import scala.annotation.tailrec
+import scala.collection.Iterator
+
+/**
+ * An object containing the red-black tree implementation used by mutable `TreeMaps`. For performance reasons, this
+ * implementation uses `null` references to represent leaves instead of a sentinel node. Currently, the internal nodes
+ * do not store their subtree size - only the tree object keeps track of their size. Therefore, while obtaining the size
+ * of the whole tree is O(1), knowing the number of entries inside a range is O(n) on the size of the range.
```

Comment:
Do you think it is worth keeping notes in the API docs about performance? Saying that it is a red-black tree tells users the complexity of most operations, and the note about size performance seems really only internal, as I do not provide a public method for getting the size of a range (`TreeMapView` uses the size of its iterator for calculating the collection size).